src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -149,218 +142,283 @@ public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
-                    uint valueB = uint.CreateTruncating(right[i]);
+                    uint valueA = uint.CreateTruncating(Unsafe.Add(ref left, i));
+                    uint valueB = uint.CreateTruncating(Unsafe.Add(ref right, i));
 
-                    if (valueA != valueB)
+                    if (!UnicodeUtility.IsAsciiCodePoint(valueA | valueB))
                     {
                         return false;
                     }
 
-                    if (!UnicodeUtility.IsAsciiCodePoint(valueA | valueB))
+                    if (valueA == valueB)
+                    {
+                        continue; // exact match
+                    }
+
+                    valueA |= 0x20u;
+                    if (valueA - 'a' > 'z' - 'a')
+                    {
+                        return false; // not exact match, and first input isn't in [A-Za-z]
+                    }
+
+                    if (valueA != (valueB | 0x20u))
                     {
                         return false;
                     }
                 }
             }
-            else if (Vector256.IsHardwareAccelerated && right.Length >= Vector256<T>.Count)
+            else if (!Vector256.IsHardwareAccelerated || length < (uint)Vector256<TRight>.Count)
             {
-                ref T currentLeftSearchSpace = ref MemoryMarshal.GetReference(left);
-                ref T oneVectorAwayFromLeftEnd = ref Unsafe.Add(ref currentLeftSearchSpace, left.Length - Vector256<T>.Count);
-                ref T currentRightSearchSpace = ref MemoryMarshal.GetReference(right);
-                ref T oneVectorAwayFromRightEnd = ref Unsafe.Add(ref currentRightSearchSpace, right.Length - Vector256<T>.Count);
+                ref TLeft currentLeftSearchSpace = ref left;
+                ref TLeft oneVectorAwayFromLeftEnd = ref Unsafe.Add(ref currentLeftSearchSpace, length - TLoader.Count128);
+                ref TRight currentRightSearchSpace = ref right;
+                ref TRight oneVectorAwayFromRightEnd = ref Unsafe.Add(ref currentRightSearchSpace, length - (uint)Vector128<TRight>.Count);
 
-                Vector256<T> leftValues;
-                Vector256<T> rightValues;
+                Vector128<TRight> leftValues;
+                Vector128<TRight> rightValues;
+
+                Vector128<TRight> loweringMask = typeof(TRight) == typeof(byte)
+                    ? Vector128.Create((byte)0x20).As<byte, TRight>()
+                    : Vector128.Create((ushort)0x20).As<ushort, TRight>();
+
+                Vector128<TRight> vecA = typeof(TRight) == typeof(byte)
+                    ? Vector128.Create((byte)'a').As<byte, TRight>()
+                    : Vector128.Create((ushort)'a').As<ushort, TRight>();
+
+                Vector128<TRight> vecZMinusA = typeof(TRight) == typeof(byte)
+                    ? Vector128.Create((byte)('z' - 'a')).As<byte, TRight>()
+                    : Vector128.Create((ushort)('z' - 'a')).As<ushort, TRight>();
```

Comment:
Can this be simplified?

```suggestion
                Vector128<TRight> loweringMask = Vector128.Create((TRight)(object)0x20);
                Vector128<TRight> vecA = Vector128.Create((TRight)(object)'a');
                Vector128<TRight> vecZMinusA = Vector128.Create((TRight)(object)('z' - 'a'));
```

Reply:
Good idea 👍🏻 
Won't work this way though, as the constants are either `int` or `char`, so an `InvalidCastException` would happen.
But it works with `TRight.CreateTruncating`.